django_app/redbox_app/redbox_core/models.py

```diff
@@ -481,3 +481,19 @@ def get_messages_ordered_by_citation_priority(cls, chat_id: uuid.UUID) -> Sequen
                 )
             )
         )
+
+
+class ChatMessageTokenUse(UUIDPrimaryKeyBase, TimeStampedModel):
+    class UseTypeEnum(models.TextChoices):
+        INPUT = "input", _("input")
+        OUTPUT = "output", _("output")
+
+    chat_message = models.ForeignKey(ChatMessage, on_delete=models.CASCADE)
```

Comment:
> What do you think about the data model in Django?

I handnt appreciated that there are many token counts per message, but this does make sense

> should ChatMessageTokenUse be exported anywhere in Django Admin (probably a new PR, tbh)

not sure about admin...  but it should definitely be included here: https://github.com/i-dot-ai/redbox/pull/990

Reply:
but , yes, another PR for sure

Reply:
> I handnt appreciated that there are many token counts per message, but this does make sense

it gives us a bit of leeway if we decide to go down a path of using different models per message, which James thought was possible.